gateway/proxy/proxy/httpserver.py

```diff
@@ -0,0 +1,159 @@
+            if len(self.raw_requestline) > 65536:
+                self.requestline = ""
+                self.request_version = ""
+                self.command = ""
+                self.send_error(HTTPStatus.REQUEST_URI_TOO_LONG)
+                logging.debug("New request: return 1")
+                return
+            if not self.raw_requestline:
+                self.close_connection = True
+                logging.debug("New request: return 2")
+                return
+            if not self.parse_request():
+                # An error code has been sent, just exit
+                logging.debug("New request: return 3")
+                return
+            if self.command == "POST":
+                self.do_POST()
+            else:
+                url = f"{dst_protocol}://" + self.headers["Host"] + self.path
+                logging.debug("Passthrough none POST request: %s", url)
+                resp = requests.request(self.command, url, headers=self.headers)
+                self.handle_response(resp)
+            self.wfile.flush()  # actually send the response if not already done.
+        except TimeoutError as e:
+            # a read or a write timed out.  Discard this connection
+            self.log_error("Request timed out: %r", e)
+            self.close_connection = True
+            return
+
+    def do_POST(self):
+        logging.debug("POST")
+        url = f"{dst_protocol}://" + self.headers["Host"] + self.path
+        logging.debug("Passthrough POST request: %s", url)
+        content_length = int(self.headers["Content-Length"])
+        data = self.rfile.read(content_length)
+        job_request = self.path.find("/runtime/jobs") != -1
+        id = None
+        if job_request:
+            if "X-Qx-Client-Application" in self.headers:
+                logging.debug(
+                    "X-Qx-Client-Application: %s",
+                    self.headers["X-Qx-Client-Application"],
+                )
+                pos = self.headers["X-Qx-Client-Application"].find("middleware_job_id")
+                if pos != -1:
+                    id = self.headers["X-Qx-Client-Application"][
+                        pos
+                        + len("middleware_job_id/") : pos
+                        + len("middleware_job_id/")
+                        + 36
+                    ]
+                    logging.debug("Middleware Job ID: %s", id)
+
+        resp = None
+        while resp == None:
+            try:
+                resp = requests.request(
+                    self.command, url, headers=self.headers, data=data
+                )
+            except:
+                None
+        self.handle_response(resp)
+        self.wfile.flush()  # actually send the response if not already done.
+        if job_request and id:
+            jsondata = json.loads(resp.content.decode("utf-8"))
+            logging.debug("job id: %s", jsondata["id"])
+
+            token = "awesome_token"
```

Comment:
Mmm... We are going to need to take a look on how to manage authentication between `gateway` and `proxy`.

Reply:
Oh, I forgot that.  Yes, we have to decide how to do it.  Thanks!